homeassistant/components/tahoma/__init__.py

```diff
@@ -42,26 +45,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     username = entry.data.get(CONF_USERNAME)
     password = entry.data.get(CONF_PASSWORD)
+    hub = entry.data.get(CONF_HUB, DEFAULT_HUB)
+    endpoint = SUPPORTED_ENDPOINTS[hub]
 
     session = aiohttp_client.async_create_clientsession(hass)
-    client = TahomaClient(username, password, session=session)
+    client = TahomaClient(
+        username,
+        password,
+        session=session,
+        api_url=endpoint,
+    )
 
     try:
         await client.login()
         devices = await client.get_devices()
         scenarios = await client.get_scenarios()
         places = await client.get_places()
     except BadCredentialsException:
-        _LOGGER.error("Invalid authentication.")
+        _LOGGER.error("Invalid authentication")
```

Comment:
```suggestion
        raise ConfigEntryAuthFailed
```

This will start a reauth flow

Reply:
@bdraco I already put this one (and some related changes) on my todo list. I did assume that we still needed some additions to the Config Flow, or is this not required anymore?

If so; I would like to do that in a follow up PR, to not clutter this one. We also left out the the zeroconf / dhcp discovery part from this PR.

Reply:
Yes you still have to implement async_step_reauth.

Reply:
@bdraco will do this in a follow up PR, together with some other minor improvements around recent core changes.